web/tests/calculate-percentage-directive-spec.js

```diff
@@ -43,13 +43,15 @@ describe('calculatePercentage directive', function() {
             this.elm = this.$compile(tpl)(this.scope);
             this.scope.$digest();
         };
+        spyOn(this.calculateCompletionPercent, 'countQuestionsTotal').and.returnValue(questionSet);
+        spyOn(this.calculateCompletionPercent, 'countNumberOfAnsweredQuestions').and.returnValue(answers);
     });
 
     it('should call this.calculateCompletionPercent.countQuestionsTotal with the questionSet', function() {
-        spyOn(this.calculateCompletionPercent, 'countQuestionsTotal').and.returnValue(questionSet);
```

Comment:
You need to call `this.compileDirective(this.template)` inside the test. You probaboy need to set  `this.scope.getResponse` and `this.scope.field`to something before compiling because otherwise the `$watch` won't trigger as the `model` and `question-set` will be empty